code/02_agg_climate_data.R

```diff
@@ -0,0 +1,240 @@
+                                    row.names=row.names, optional=optional, xy=xy, ...)))
+      v <- rbindlist(l)
+    }
+    coln <- names(x)
+    if(xy) coln <- c("x", "y", coln)
+    setnames(v, coln)
+    v
+  }
+  setMethod('as.data.table', signature(x='Raster'), as.data.table.raster)
+  
+  ## Load saved weights 
+  ## -----------------------------------------------
+  
+  # Normalize the data source input - remove spaces & lower case
+  data_source_norm <- gsub(" ", "", data_source) %>% tolower(.)
+  
+  # File name to call the weights
+  if(weights){
+    weights_file <- paste0(paste(input_polygons_name, data_source_norm, "area", weights_type, "weights", sep="_"), ".csv")
+  } else{
+    weights_file <- paste0(paste(input_polygons_name, data_source_norm, "area", "weights", sep="_"), ".csv")
+  }
+  
+  # Data.table of weights 
+  weights_dt <- fread(file.path(here::here(), "data", "int", "weights", weights_file))
+  # Convert 0-360 to match climate raster
+  weights_dt[, x := ifelse(x < 0, x + 360, x)]
+  
+  # Extent of area weights with slight buffer to make sure all cells are included 
+  min_x <- min(weights_dt$x) - 0.5
+  max_x <- max(weights_dt$x) + 0.5
+  min_y <- min(weights_dt$y) - 0.5
+  max_y <- max(weights_dt$y) + 0.5
+  
+  weights_ext <- raster::extent(min_x, max_x, min_y, max_y)
+  
+  ## Load climate data 
+  ## -----------------------------------------------
+  
+  # Check if the climate variable is one we have data for 
+  if(!climate_var %in% c('prcp', 'temp', 'uv')){
+    stop(crayon::red('No ERA5 data available. Supported variables are: prcp, temp, or uv'))
+  }
+  
+  # Climate data file paths
+  ncpath  <- file.path(here::here(), 'data/raw', climate_var)
+  ncname  <- paste(data_source_norm, climate_var, year, sep="_")
+  nc_file <- paste0(ncpath, '/', ncname,'.nc')
+  
+  # Immediately crop to weights extent 
+  clim_raster <- raster::crop(raster::stack(nc_file), weights_ext)
+  
+  # Get layer names (dates)
+  all_layers <- names(clim_raster)
+  layer_names <- all_layers[seq(1, length(all_layers), 24)] # Keep every 24th layer name (1 per day)
+  layer_names <- paste0('month_', substring(layer_names, 7,8), '_day_', substring(layer_names, 10,11)) # Extract month/day for each layer 
+  
+  ## Aggregate to grid-day level  
+  ## -----------------------------------------------
+  
+  if(!daily_agg %in% c('average', 'sum')){
+    
+    stop(crayon::red("Daily aggregation must be 'average' or 'sum'"))
+  }
+  
+  ## Average 
+  if(daily_agg == 'average'){
+    
+    # Average over each set of 24 layers - assuming there are 24*365 layers 
+    indices<-rep(1:(nlayers(clim_raster)/24),each=24)
+    clim_daily <- raster::stackApply(clim_raster, indices = indices, fun=mean) #Stack of 365 layers
+  }
+
+  ## Sum 
+  if(daily_agg == 'sum'){
+    
+    # Sum over each set of 24 layers - assuming there are 24*365 layers
+    indices<-rep(1:(nlayers(clim_raster)/24),each=24)
```

Comment:
@saraorofino same here on the assumption

Reply:
Done